src/main/java/com/stuypulse/robot/subsystems/Conveyor.java

```diff
@@ -42,96 +39,130 @@
+        topIRSensor = new DigitalInput(Ports.Conveyor.TOP_CONVEYOR_IR_SENSOR);
+        shooting = false;
     }
 
-    /**
-     * Spins the Top Conveyor Belt, moving the ball up to the shooter
-     */
-    public void spinTopBelt() {
+    public boolean isShooting() {
+        return shooting;
+    }
+
+    public void setShoot(boolean shooting) {
+        this.shooting = shooting;
+    }
+
+    /** Spins the Top Conveyor Belt, moving the ball up to the shooter */
+    private void spinTopBelt() {
         topMotor.set(ConveyorSettings.TOP_MOTOR_SPEED.get());
     }
-    
+
     /**
-     * Accept ball - spin the gandalf motor upwards to the top conveyor
-     * To be used when the ball is team alliance color
+     * Accept ball - spin the gandalf motor upwards to the top conveyor To be used when the ball is
+     * team alliance color
      */
-    public void acceptBall() {
+    private void acceptBall() {
         gandalfMotor.set(ConveyorSettings.ACCEPT_SPEED.get());
     }
-    
+
     /**
-     * Eject ball - spin the gandalf motor outwards
-     * To be used when the ball is opposing alliance color
+     * Eject ball - spin the gandalf motor outwards To be used when the ball is opposing alliance
+     * color
      */
-    public void rejectBall(){
+    private void rejectBall() {
         // If the ball is not of our alliance color, reject ball
         gandalfMotor.set(ConveyorSettings.REJECT_SPEED.get());
     }
-    
-    /**
-     * Stops the Top Conveyor Belt
-     */
-    public void stopTopBelt() {
+
+    /** Stops the Top Conveyor Belt */
+    private void stopTopBelt() {
         topMotor.stopMotor();
     }
 
-    /**
-     * Stops the Gandalf Motor
-     */
-    public void stopGandalf() {
+    /** Stops the Gandalf Motor */
+    private void stopGandalf() {
         gandalfMotor.stopMotor();
     }
 
-    /**
-     * Stop both the Top Conveyor Belt and the Gandalf Motor
-     */
-    public void stop() {
-        stopTopBelt();
-        stopGandalf();
+    /** Finds if the upper IR Sensor has been tripped e.g., there is a ball in the top conveyor */
+    private boolean getTopConveyorUpperHasBall() {
+        return topIRSensor.get();
     }
 
-    /**
-     * Gets the CurrentBall object from the color sensor
-     * @return CurrentBall detected by the color sensor
-     */
-    private CurrentBall getCurrentBall() {
-        return colorSensor.getCurrentBall();
+    /** Finds if the lower IR Sensor has been tripped */
+    private boolean getTopConveyorLowerHasBall() {
+        return gandalfIRSensor.get();
     }
-    
-    /**
-     * Finds if the upper IR Sensor has been tripped e.g., there is a ball in the top conveyor
-     * @return if the upper IR Sensor has been tripped
-     */
-    public boolean getTopConveyorHasBall() {
-        return irSensor.get();
+
+    private boolean getEjecting() {
+        return colorSensor.hasOpponentBall();
+    }
+
+    private boolean getRunning() {
```

Comment:
`getRunning()` and `getEjecting()` are kinda difficult names and they hide the idea that they represent. It might be worth it to think about what we mean with these variables.